tests/unit/request_mock.py

```diff
@@ -0,0 +1,101 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional
+
+import pytest
+
+
+class MockRequestsResponse:
```

Comment:
It was named this way because it mocks the Response class from the requests package, but I agree this is confusing. Going to figure new names for it and RequestMock.